ert3/engine/_record.py

```diff
@@ -47,4 +47,5 @@ def sample_record(
         record_name=record_name,
         ensemble_record=ensrecord,
         experiment_name=experiment_name,
+        is_parameter=True,
```

Comment:
Why does this have to leak out here? Is this not double configuration in the sense that the record is already registered as a parameter upon initialization of experiment?

Reply:
ERT Storage is informed of the record containing parameters only once. So in that sense it's not a double configuration.

The alternative is to ask ERT Storage again, which is adds an unnecessary request, or to store the information in a global variable. We already know it's a parameter at this point, so being specific at this point is fine. Also we avoid a O(log _n_) to determine if it's a parameter.

Reply:
Agree with Zohar, but also understand that `is_parameter` seems like providing the same information twice. What about making it more general specifying `is_compound=True` (my favorite word these days! :) ) or just `do_split=True`, meaning this record becomes split into split into several ones :)

Reply:
I don't like the word "compound" wrt. parameters since it creates this separation between "parameter" and "compound parameter". We don't need this complexity. What you're calling a "compound parameter" I call a "parameter record", ie. a record that contains one or more parameters.

In my naming convention, both ERT2 and ERT3 publish "parameter records". ERT3 contains multiple parameters within the records, while ERT2 only contains 1 per record. ERT Storage only lets you fetch whole "parameter records". If you want to fetch a single parameter, we require you to fetch the entire record and select the column you want.

Using "compound parameter" kind of makes it sound like ERT Storage needs to consider "parameter" records and "compound parameter" records as separate types, which separate logic for each.

Reply:
Two comments here :)

First, there is a discrepancy between the notion of a record and a parameter in ert3 and ert-storage. The responsibility of handling this discrepancy should reside entirely on `ert3.storage`. Afterwards, when we have working software, we should align on the language and definition of a record and parameter. But as long as within the `ert3` module (besides `ert3.storage`), the `ert3.data.Record` should serve as the only definition of what a record is. And there there is no "compound" nor "parameter" record present there.

Second, upon initialising the experiment `ert3` is passing the list of names of all the parameters (or now a dict also containing the keys). That contains the information needed and since the notion of a parameter is not really well-defined in `ert3` I would prefer to keep it as much as possible out of the implementation. In particular, it is only used in the `export` code for now... Also, if we move in the direction currently suggested in https://github.com/equinor/ert/issues/1731 this distinction between a "parameter" or not a "parameter" should disappear again very soon. And then it is also nice to keep the current introduction of the concept as local as possible (currently only for initialising experiment and export).

Reply:
The distinction between what a parameter and non-parameters will still be an issue even with #1731 . In fact, ERT Storage already uses a similar format for matrix data (what you call "numerical data"). This distinction is very relevant for webviz-ert, which needs to have logic for visualising parameters, which necessitates specifying which records contain parameter data (and then also tying them to a parameter prior distribution if it exists).